pkg/gtest/gocovertest.go

```diff
@@ -0,0 +1,97 @@
+package gtest
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/gocover/pkg/gittool"
+)
+
+const testFileSuffix = "_test.go"
+
+type GocoverTest struct {
+	RepositoryPath string
+	CompareBranch  string
+	GitClient      gittool.GitClient
+	Writer         io.Writer
+}
+
+func NewGocoverTest(
+	repositoryPath string,
+	compareBranch string,
+	writer io.Writer,
+) (*GocoverTest, error) {
+
+	gitClient, err := gittool.NewGitClient(repositoryPath)
+	if err != nil {
+		return nil, fmt.Errorf("git repository: %w", err)
+	}
+
+	return &GocoverTest{
+		RepositoryPath: repositoryPath,
+		CompareBranch:  compareBranch,
+		GitClient:      gitClient,
+		Writer:         writer,
+	}, nil
+}
+
+func (g *GocoverTest) CheckGoTestFiles() error {
+	changes, err := g.GitClient.DiffChangesFromCommitted(g.CompareBranch)
+	if err != nil {
+		return fmt.Errorf("git diff: %w", err)
+	}
+
+	for _, c := range changes {
+		fmt.Println(c.FileName)
+		folder := filepath.Dir(filepath.Join(g.RepositoryPath, c.FileName))
+
+		fmt.Fprintf(g.Writer, "checking test files for package: %s\n", folder)
+		exist, err := checkTestFileExistence(folder)
+		if err != nil {
+			return fmt.Errorf("check test file existence: %w", err)
+		}
+		if exist {
+			continue
+		}
+
+		_, pkgName := filepath.Split(folder)
+		testFile := filepath.Join(folder, testFileName(pkgName))
```

Comment:
may need the "package pkgname" in the header?

Reply:
What's header? Previous `Dir` function will return the folder of the go file. And After split, it only leaves the package name.

Reply:
the dir name may be different with the packge name.

Reply:
parse package name from an existing go file under the folder.